app/Jobs/WikiEntityImportJob.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+namespace App\Jobs;
+
+use App\WikiEntityImportStatus;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\Config;
+use App\Wiki;
+use App\WikiEntityImport;
+use Carbon\Carbon;
+use Maclof\Kubernetes\Client;
+use Maclof\Kubernetes\Models\Job as KubernetesJob;
+
+class WikiEntityImportJob implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    /**
+     * Create a new job instance.
+     */
+    public function __construct(
+        public int $wikiId,
+        public string $sourceWikiUrl,
+        public array $entityIds,
+        public int $importId,
+    )
+    {}
+
+    private string $targetWikiUrl;
+
+    /**
+     * Execute the job.
+     */
+    public function handle(Client $kubernetesClient): void
+    {
+        $import = null;
+        try {
+            $wiki = Wiki::findOrFail($this->wikiId);
+            $import = WikiEntityImport::findOrFail($this->importId);
+            $creds = $this->acquireCredentials($wiki->domain);
+
+            $this->targetWikiUrl = str_contains($wiki->domain, "localhost")
```

Comment:
This is ugly. I'd be happy to learn about a cleaner solution for this.

Reply:
only suggestion I have is we revisit the behaviour of the TransferBot image *again* and do something like the sidecar proxy thing and always talk to that over http.

I agree this is a nasty leaking of knowing the details about the network environment of the job container into the api.

We could at least tighten it by parsing the url and checking it's the TLD of the FQDN. That would save us having some mystery errors in the future for `localhostingofdinnerparties.wikibase.cloud` which this would inadvertently catch

Reply:
Yeah, I think parsing the domain and then checking its TLD is a good compromise.